op-deployer/pkg/deployer/apply.go

```diff
@@ -343,6 +357,16 @@ func ApplyPipeline(
 		})
 	}
 
+	if opts.PreStateBuilder != nil {
```

Comment:
I think the warning should move to in here. Other libraries can call into the pipeline so the warning wouldn't show up in those cases.

Reply:
> Other libraries can call into the pipeline so the warning wouldn't show up in those cases.

Fair enough, will do. 

Though it will feels weird for apply.go to expose potentially shared functionality with other libs. I am getting more and more inclined towards a "pipeline"-dedicated package serving such shared functionalities. Can definitely explore that in a separate PR.